### PR TITLE
[TEST - DO NOT MERGE] Switch to clang 3.9 as the default

### PR DESCRIPTION
**Pull request by janvorli**

### CODE REVIEW

**Comment by janvorli** [comment]
CC: @gkhanna79 here is the testing PR, let's see the results of the Ubuntu tests. The other Linux distros probably do not have the clang 3.9 installed in the CI.

**Comment by janvorli** [comment]
@mmitche is the coreclr CI lab using different machines from the corert one? I was assuming it is the same and so I have submitted this testing PR to try to set the default clang to 3.9 like we do in CoreRT, but the Ubuntu x64 builds are failing due to the missing clang.
If this lab is different, could you please install clang 3.9 on the Ubuntu machines for now so that we can do some testing before we move to that on all linux distros?

**Comment by mmitche** [comment]
@janvorli It's certainly possible.   I would check the version field on the setMachineAffinity calls for the two repos.

**Comment by hqueue** [comment]
FYI. All four CI for arm cross build (including arm and armel) fails, because docker image used in CI  don't have clang 3.9 installed.

**Comment by janvorli** [comment]
@hqueue thank you for the notice. I am aware of that, I wanted to test just the Ubuntu x64.

**Comment by janvorli** [comment]
@mmitche I can see that in the coreclr netci.groovy, the machine affinity for all x64 distros is set this way:
```groovy
Utilities.setMachineAffinity(job, os, 'latest-or-auto');
```
But on CoreRT, it is set like this:
```groovy
Utilities.setMachineAffinity(newJob, os, imageVersionMap[os])
```
where imageVersionMap['Ubuntu']='20170118'

What would be the best way to change it? Should I just edit the function below by adding a special case for Ubuntu x64 and set the version to '20170118' for it?
And to test it, can I test the change to the netci.groovy and my change to coreclr build in a single PR or do I need to merge in the change to netci.groovy first? I think it was that way in the past, but I have a feeling that I've heard somewhere that there were some changes in this area.
```groovy
def static setMachineAffinity(def job, def os, def architecture) {
    if (architecture == 'arm64' && os == 'Windows_NT') {
        // For cross compilation
        job.with {
            label('arm64')
        }
    } else if (architecture == 'arm64' && os == 'Ubuntu') {
        Utilities.setMachineAffinity(job, os, 'arm-cross-latest');
    } else if ((architecture == 'arm') && (os == 'Ubuntu' || os == 'Ubuntu16.04' || os == 'Tizen')) {
        Utilities.setMachineAffinity(job, 'Ubuntu', 'arm-cross-latest');
    } else {
        Utilities.setMachineAffinity(job, os, 'latest-or-auto');
    }
}
```

**Comment by mmitche** [comment]
@janvorli You can add a special case for this for now. You should change that, do the ci test command, then we can enable one of the generated jobs and see what happens.

FYI, you're going to need to change more than just 14.04 I think.  16.04, the tizen builds, and freebsd don't use that same 14.04 image.

What about official, do we have 3.9 in the docker containers?

**Comment by janvorli** [comment]
@mmitche thank you for your reply. This is just a first step to verify that everything including CoreCLR tests pass correctly in the CI. After that, we would like to enable 3.9 everywhere, it means for all distros and their versions, cross build docker containers, FreeBSD, Tizen, ...

File: build.sh [hunk 1]
```diff
@@ -781,10 +781,10 @@ fi
 if [[ $__ClangMajorVersion == 0 && $__ClangMinorVersion == 0 ]]; then
     if [ $__CrossBuild == 1 ]; then
         __ClangMajorVersion=3
-        __ClangMinorVersion=6
+        __ClangMinorVersion=9
     else
         __ClangMajorVersion=3
-        __ClangMinorVersion=5
+        __ClangMinorVersion=9
```
Comment (rartemev) [inline build.sh]: Is it normal that both "if" branches are equal? I think it would be good to delete inner condition "if [ $__CrossBuild == 1 ];"

Comment (janvorli) [inline build.sh]: This is just a testing PR, not a final PR for the switch. You are right of course that after we make the change, we would remove the if / else.

**Pull request closed without merging**